tests/test_modeling_encoder_decoder.py

```diff
@@ -0,0 +1,177 @@
+# coding=utf-8
+# Copyright 2018 Google T5 Authors and HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import unittest
+import tempfile
+import os
+
+from transformers import is_torch_available
+
+from .utils import require_torch, slow
+
+# this line reruns all the tests in BertModelTest; not sure whether this can be prevented
+# for now only run module with pytest tests/test_modeling_encoder_decoder.py::EncoderDecoderModelTest
+from .test_modeling_bert import BertModelTest
```

Comment:
This was quite annoying. As far as I can see it is not possible to import a unittest testcase, such as `BertModelTest from .test_modeling_bert` without running all of its test. At the same time I think we need to import the TestCase here since we need to instantiate BertEncoder and BertDecoder models without copy pasting all of its code. I tried a couple of hacks, but didn't find a good solution. Maybe it's possible to only run `pytest tests/test_modeling_encoder_decoder.py::EncoderDecoderModelTest` from cicrle ci in which case only the encoder-decoder model tests are run? Any good ideas for this case? @julien-c @LysandreJik @thomwolf @sshleifer

Reply:
1) Move the ModelTester to the top level of that file.
2) in that file, `self.model_tester = BertModelTester(self)`

3) only import `BertModelTester`

verified that no extra tests are run.

You can merge https://github.com/huggingface/transformers/pull/4027 if you want to keep your delta small.

Reply:
Awesome! I will try this out :-)

Reply:
Would it be ok to change Bert's testing structure as shown in PR #4027 for you guys @julien-c @thomwolf @LysandreJik ?